android/app/src/main/kotlin/net/mullvad/mullvadvpn/compose/dialog/DnsDialog.kt

```diff
@@ -63,35 +76,62 @@ private fun PreviewDnsDialogEdit() {
 private fun PreviewDnsDialogEditAllowLanDisabled() {
     AppTheme {
         DnsDialog(
-            stagedDns =
-                StagedDns.EditDns(
-                    CustomDnsItem(address = "1.1.1.1", isLocal = true),
-                    StagedDns.ValidationResult.Success,
-                    0
-                ),
-            isAllowLanEnabled = false,
-            onIpAddressChanged = {},
-            onAttemptToSave = {},
-            onRemove = {},
-            onDismiss = {}
+            DnsDialogViewState(
+                "192.168.1.1",
+                DnsDialogViewState.ValidationResult.Success,
+                true,
+                false,
+                true
+            ),
+            {},
+            {},
+            {},
+            {}
         )
     }
 }
 
+@Destination(style = DestinationStyle.Dialog::class)
 @Composable
 fun DnsDialog(
-    stagedDns: StagedDns,
-    isAllowLanEnabled: Boolean,
-    onIpAddressChanged: (String) -> Unit,
-    onAttemptToSave: () -> Unit,
-    onRemove: () -> Unit,
+    resultNavigator: ResultBackNavigator<Boolean>,
+    index: Int?,
+    initialValue: String?,
```

Comment:
Long-term I think it would be nice to break up some of our dialogs to NewDnsDialog and ModifyDnsDialog, then we can avoid having null as arguments and conditional rendering with `isNewEntry`